File: Leetcode/Leetcode-DSA-course/Hashing/560-subarray-sum-equals-k/subarray-sum-equals-k.ts
function subarraySum(nums: number[], k: number): number {
    const counts = new Map();
    counts.set(0, 1);
    let ans = 0, curr = 0;
    for (let num of nums) {
        curr += num;
        ans += counts.get(curr - k) || 0;
        counts.set(curr, (counts.get(curr) || 0) + 1)
    }
    return ans;
};

File: Leetcode/Leetcode-DSA-course/Hashing/560-subarray-sum-equals-k/subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        Map<Integer, Integer> counts = new HashMap<>();
        counts.put(0, 1);
        int ans = 0, curr = 0;
        for (int num : nums) {
            curr += num;
            ans += counts.getOrDefault(curr - k, 0);
            counts.put(curr, counts.getOrDefault(curr, 0) + 1);
        }
        return ans;
    }
}